src/app/inventory/item-move-service.ts

```diff
@@ -554,71 +560,80 @@ function ItemService(): ItemServiceType {
 
     // Find any stackable that could be combined with another stack
     // on a different store to form a single stack
-    if (item.maxStackSize > 1) {
-      let otherStore: DimStore | undefined;
-      const stackable = moveAsideCandidates.find((i) => {
-        if (i.maxStackSize > 1) {
-          // Find another store that has an appropriate stackable
-          otherStore = otherStores.find((s) =>
-            s.items.some(
-              (otherItem) =>
-                // Same basic item
-                otherItem.hash === i.hash &&
-                !otherItem.location.inPostmaster &&
-                // Enough space to absorb this stack
-                i.maxStackSize - otherItem.amount >= i.amount
-            )
-          );
-        }
-        return Boolean(otherStore);
-      });
-      if (stackable && otherStore) {
-        return {
-          item: stackable,
-          target: otherStore
-        };
+    let otherStore: DimStore | undefined;
+    const stackable = moveAsideCandidates.find((i) => {
+      if (i.maxStackSize > 1) {
+        // Find another store that has an appropriate stackable
+        otherStore = otherStores.find((s) =>
+          s.items.some(
+            (otherItem) =>
+              // Same basic item
+              otherItem.hash === i.hash &&
+              !otherItem.location.inPostmaster &&
+              // Enough space to absorb this stack
+              i.maxStackSize - otherItem.amount >= i.amount
+          )
+        );
       }
+      return Boolean(otherStore);
+    });
+    if (stackable && otherStore) {
+      return {
+        item: stackable,
+        target: otherStore
+      };
     }
 
     const tierValue = {
       Common: 0,
       Uncommon: 1,
       Rare: 2,
-      Legendary: 3,
-      Exotic: 4
+      Legendary: 4,
+      Exotic: 3
```

Comment:
I also inverted the priority of legendaries and exotics, to avoid characters getting loaded up with exotics.